### PR TITLE
int-test: Initial Support for Containers + persistentdw server fix

### PR DESCRIPTION
**Pull request by bdevcich-hpe**
Add in two Pending container tests and the required changes necessary to
run container tests with container profiles.

In order to test containers with persistent storage, a check was added
Assign Servers that skips the block when a `persisentdw` directive is
present.

Both tests have been successful in kind with the `feature/containers`
branch.

Signed-off-by: Blake Devcich <blake.devcich@hpe.com>

### CODE REVIEW

File: test/internal/options.go [hunk 1]
```diff
@@ -172,16 +194,40 @@ func (t *T) Prepare(ctx context.Context, k8sClient client.Client) error {
 		Expect(k8sClient.Create(ctx, profile)).To(Succeed())
 	}
 
+	if o.containerProfile != nil {
+		// Clone the placeholder profile
```
Comment (NateThornton) [inline test/internal/options.go:198]: Why do we need to clone it? Wouldn't it make more sense to specify the existing container name so you know what you're getting ("example-success") up front, in the test definition?

Comment (NateThornton) [reply test/internal/options.go:198]: Now I'm wondering why I didn't just use the placeholder profile for storage profiles too...

File: test/internal/options.go [hunk 2]
```diff
@@ -172,16 +194,40 @@ func (t *T) Prepare(ctx context.Context, k8sClient client.Client) error {
...
 	}
 
 	if o.persistentLustre != nil {
 		// Create a persistent lustre instance all the way to pre-run
 		name := o.persistentLustre.name
+		capacity := o.persistentLustre.capacity
 
 		o.persistentLustre.create = MakeTest(name+"-create",
-			fmt.Sprintf("#DW create_persistent type=lustre name=%s capacity=1TB", name))
+			fmt.Sprintf("#DW create_persistent type=lustre name=%s capacity=%s", name, capacity))
```
Comment (NateThornton) [inline test/internal/options.go:230]: I don't think there's any need to customize this with a test parameter. Just make it smaller 100GB

File: test/int_test.go [hunk 3]
```diff
@@ -77,15 +77,26 @@ var tests = []*T{
...
 		WithGlobalLustreFromPersistentLustre("/lus/global").
 		Serialized().
 		Pending(),
 
-	MakeTest("GFS2 with Containers (BLAKE)",
-		"#DW jobdw type=gfs2 name=gfs2-with-containers capacity=1TB",
-		"#DW container name=gfs2-with-containers profile=TODO DW_JOB_gfs2-with-containers",
-	).Pending(),
+	// Containers
+	// TODO: requires nnf-sos deployed with `feature/containers` branch
```
Comment (NateThornton) [inline test/int_test.go:86]: This also requires containers are in the nnf-sos submodule. That's why the build is failing

File: test/internal/options.go (hunk 1 above)
Comment (bdevcich-hpe) [reply test/internal/options.go]: I removed this and turned to using the examples.

File: test/internal/options.go (hunk 2 above)
Comment (bdevcich-hpe) [reply test/internal/options.go:186]: Done

File: test/internal/states.go [hunk 4]
```diff
@@ -94,10 +95,15 @@ func (t *T) setup(ctx context.Context, k8sClient client.Client, workflow *dwsv1a
 				return directiveBreakdown.Status.Ready
 			}).Should(BeTrue())
 
+			// persistentdw directives do not have StorageBreakdowns (Status.Storage)
```
Comment (NateThornton) [inline test/internal/states.go:98]: Should this be an Expect() clause itself?

**Review by NateThornton — APPROVED** [review]

**Pull request merged**